src/renderer/services/bitcoin/ledger.ts

```diff
@@ -18,10 +22,53 @@ const retrieveLedgerAddress = (network: Network) =>
     catchError((error) => Rx.of(RD.failure(error)))
   ).subscribe(setLedgerAddressRD)
 
-const createLedgerService = (): LedgerService => ({
+export const createLedgerService = (): LedgerService => ({
   ledgerAddress$,
   retrieveLedgerAddress,
   removeLedgerAddress: () => setLedgerAddressRD(RD.initial)
 })
 
-export { createLedgerService }
+export const createLedgerTransferService = <Client extends XChainClient, T extends XChainClient$<Client>>(
+  client$: T
+) => {
+  const { get$: ledgerTxRD$, set: setLedgerTxRD } = observableState<TxRD>(RD.initial)
+
+  /**
+   * According to the Client's interface
+   * first argument type of Client.transfer method
+   * is an object of TxParams.
+   * @see https://github.com/xchainjs/xchainjs-lib/blob/master/packages/xchain-client/src/types.ts#L96
+   *
+   *  In common-client case
+   * this parameters might be extended so we need this generic type
+   * to have an access to params "real" type value for specific chain
+   * @example BTC client has extended TxParams interface
+   * @see https://github.com/xchainjs/xchainjs-lib/blob/master/packages/xchain-bitcoin/src/client.ts#L396
+   */
+  type TransferParams = Parameters<Client['transfer']>[0]
+
+  const ledgerTx$ = (params: TransferParams): TxLD =>
+    client$.pipe(
+      RxOp.switchMap((oClient) => (O.isSome(oClient) ? Rx.of(oClient.value) : Rx.EMPTY)),
+      RxOp.switchMap((client) => Rx.from(client.transfer(params))),
+      RxOp.map(RD.success),
+      RxOp.catchError(
+        (e): TxLD =>
+          Rx.of(
+            RD.failure({
+              msg: e.toString(),
+              errorId: ErrorId.SEND_TX
+            })
+          )
+      ),
+      RxOp.startWith(RD.pending)
+    )
```

Comment:
No need to duplicate code of `tx$` of `src/renderer/services/clients/transaction/transfer.ts` here.

Try to create a stream using existing `tx$()` function and use it to update state with `setLedgerTxRD`. If you think it's easier to do it in `transfer.ts`, just go for it. It's still fine to have `ledgerXYZTransfer` functions in `transfer.ts`.

Reply:
sec, this is not completed code, let's discuss about this in slack after I update this code